.github/workflows-source/ci-main.yml

```diff
@@ -158,32 +157,16 @@ jobs:
     timeout-minutes: 120
     runs-on:
       labels: macOS
-    if: ${{ github.event_name != 'merge_group' }}
+    if: ${{ github.ref_protected }}
     steps:
       - <<: *checkout
-      - name: Filter Relevant Files
```

Comment:
Can you clarify why this is removed?

_edit:_ looks like this one now would only run on master and hence not worth the optimization

Reply:
Exactly, that filtering logic was only applied to PRs so it's obsolete now.